### DISCUSSION TITLE
Use DataRow for specific TestCategory

### DISCUSSION THREAD

**Discussion by OlegKalinin** [microsoft/testfx#3778] [Q&A]
Hi All!
I have data-driven test in my test solution like

```
[TestMethod]
[TestCategory("Common")]
[DataRow(1, 1, 2)]
[DataRow(2, 2, 4)]
[DataRow(3, 3, 6)]
[DataRow(0, 0, 1)] 
public void AddIntegers_FromDataRowTest(int x, int y, int expected)
...
```

What approach do you use if want to define [TestCategory("Smoke")] that should include only one DataRow of AddIntegers_FromDataRowTest test function with current test framework architecture?

I think about 
```
[TestMethod]
[TestCategory("Smoke")]
[DataRow(1, 1, 2)]
public void AddIntegers_FromDataRowTest_Smoke(int x, int y, int expected)
{
  this.AddIntegers_FromDataRowTest(x,y,expected);
}
```
but may be there is something better

**Comment by Evangelink** [comment]
Hi @OlegKalinin,

Do you want to be applying a custom category to only some of the data input? If so, that's currently not supported outside of the solution you are describing above.

I would need to make some tests to see how TRX/VS would handle that but it seems something valuable to be able to add metadata to some of the input.

**Reply by Evangelink** [reply]
We now have #4089 to track this.
